### PR TITLE
Fixes part of #713 : Lowfi Landscape Topic Practice

### PR DESCRIPTION
**Pull request by Sarthak2601**
## Explanation

- Fixes part of #713 
- Landscape mode of Topic practice according to the mock. 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by rt4914 — CHANGES_REQUESTED** [review]
You need to added test cases to check whether the landscape mode is working correctly or not. You will need to do that in `TopicPracticeFragmentTest` file. You can refer to following files for landscape test cases:
`AdminAuthActivityTest`
`RecentlyPlayedFragmentTest`

**Comment by Sarthak2601** [comment]
> You need to added test cases to check whether the landscape mode is working correctly or not. You will need to do that in `TopicPracticeFragmentTest` file. You can refer to following files for landscape test cases:
> `AdminAuthActivityTest`
> `RecentlyPlayedFragmentTest`

Okay, working on it.

**Comment by Sarthak2601** [comment]
@rt4914 PTAL :)

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 1]
```diff
@@ -194,6 +194,21 @@ class TopicPracticeFragmentTest {
...
+  @Test
+  fun testTopicPracticeFragment_loadFragment_configurationChange_TitleisCorrect() {
+    launchTopicActivityIntent(FRACTIONS_TOPIC_ID).use {
+      onView(
+        allOf(
+          withText(TopicTab.getTabForPosition(2).name),
+          isDescendantOfA(withId(R.id.topic_tabs_container))
+        )
+      ).perform(click())
+      onView(isRoot()).perform(orientationLandscape())
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:211]: Use this same code in this entire file for all other test cases too, meaning:

Replace this
```
activityScenario.onActivity { activity ->
      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
}
```


with

```
onView(isRoot()).perform(orientationLandscape())
```

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 2]
```diff
@@ -194,6 +194,21 @@ class TopicPracticeFragmentTest {
     onView(withId(R.id.topic_practice_start_button)).check(matches(isClickable()))
   }
 
+  @Test
+  fun testTopicPracticeFragment_loadFragment_configurationChange_TitleisCorrect() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt]: Naming convention is incorrect, the first character after `_` should be small.

`testTopicPracticeFragment_loadFragment_configurationChange_titleisCorrect `

**Review by rt4914 — CHANGES_REQUESTED** [review]
Some changes suggested.

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 2 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt]: Done :)

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt (hunk 1 above)
Comment (Sarthak2601) [reply app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:211]: Done :)

**Comment by Sarthak2601** [comment]
@rt4914 PTAL :)

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 3]
```diff
@@ -160,40 +163,57 @@ class TopicPracticeFragmentTest {
...
-  @Ignore("Landscape not properly supported") // TODO(#56): Reenable once landscape is supported.
   fun testTopicPracticeFragment_loadFragment_selectSkills_configurationChange_skillsAreSelected() {
-    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(click())
-    activityScenario.onActivity { activity ->
-      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
-    }
-    activityScenario.recreate()
-    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).check(matches(isChecked()))
+    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(
+      click()
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:168]: Write it in a single line like this:
`    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(click())`

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 4]
```diff
@@ -160,40 +163,57 @@ class TopicPracticeFragmentTest {
...
+    )
+    onView(isRoot()).perform(orientationLandscape())
+    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).check(
+      matches(isChecked())
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:172]: write this in a single line:
`onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).check(matches(isChecked()))`

File: app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt [hunk 5]
```diff
@@ -160,40 +163,57 @@ class TopicPracticeFragmentTest {
...
-  @Ignore("Landscape not properly supported") // TODO(#56): Reenable once landscape is supported.
   fun testTopicPracticeFragment_loadFragment_selectSkills_configurationChange_startButtonRemainsActive() {
-    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(click())
-    activityScenario.onActivity { activity ->
-      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
-    }
-    activityScenario.recreate()
-    onView(withId(R.id.topic_practice_skill_list)).perform(scrollToPosition<RecyclerView.ViewHolder>(10))
+    onView(atPositionOnView(R.id.topic_practice_skill_list, 1, R.id.subtopic_check_box)).perform(
+      click()
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/practice/TopicPracticeFragmentTest.kt:191]: Write it in a single line

**Review by rt4914 — CHANGES_REQUESTED** [review]
Please make sure your test cases are passing before assigning it for review. Your all 3 test cases are failing.

**Comment by Sarthak2601** [comment]
> Please make sure your test cases are passing before assigning it for review. Your all 3 test cases are failing.

Actually I just wrote one in there. The rest were already typed but ignored using the annotation. But yes, this is a mistake from my end. Will rectify in future.

**Comment by rt4914** [comment]
> > Please make sure your test cases are passing before assigning it for review. Your all 3 test cases are failing.
> 
> Actually I just wrote one in there. The rest were already typed but ignored using the annotation. But yes, this is a mistake from my end. Will rectify in future.

I understand this, but those tests were written way back and after that the implementation has changed. And because earlier we were not supporting landscape thats why they have not been updated and therefore they need to be updated in this PR.

**Comment by Sarthak2601** [comment]
Closing this PR and opening a fresh one.

**Pull request closed without merging**